configs/components/curl.rb

```diff
@@ -4,18 +4,45 @@
   pkg.url "http://buildsources.delivery.puppetlabs.net/curl-#{pkg.get_version}.tar.gz"
 
   pkg.build_requires "openssl"
+  pkg.build_requires "puppet-ca-bundle"
+
+  prefix = settings[:prefix]
+  pem_file = "#{settings[:prefix]}/ssl/cert.pem"
+
+  make = platform[:make]
+
+  if platform.is_windows?
+    pkg.build_requires "pl-zlib-#{platform.architecture}"
+    pkg.build_requires "runtime"
+
+    arch = platform.architecture == "x64" ? "64" : "32"
+    pkg.environment "PATH" => "#{settings[:gcc_bindir]}:$$PATH"
+    pkg.environment "CYGWIN" => settings[:cygwin]
+    pkg.environment "CC" => settings[:cc]
+    pkg.environment "CXX" => settings[:cxx]
+
+    prefix = platform.convert_to_windows_path(settings[:prefix])
+    pem_file = platform.convert_to_windows_path("#{settings[:puppet_configdir]}/ssl/cert.pem")
+
+    make = "/usr/bin/make"
+  end
 
   pkg.configure do
     ["LDFLAGS='#{settings[:ldflags]}' \
-     ./configure --prefix=#{settings[:prefix]} \
-     --with-ssl=#{settings[:prefix]} --enable-threaded-resolver --disable-ldap --disable-ldaps --with-ca-bundle=#{settings[:prefix]}/ssl/cert.pem"]
+     ./configure --prefix=#{prefix} \
+        --with-ssl=#{prefix} \
+        --enable-threaded-resolver \
+        --disable-ldap \
+        --disable-ldaps \
+        --with-ca-bundle=#{pem_file} \
+        #{settings[:host]}"]
   end
 
   pkg.build do
-    ["#{platform[:make]} -j$(shell expr $(shell #{platform[:num_cores]}) + 1)"]
+    ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1)"]
   end
 
   pkg.install do
-    ["#{platform[:make]} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
+    ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
```

Comment:
If a different make is being used for all components in the agent, it would be worth setting `platform.make "/usr/bin/make"` in the windows platform config. Then you can avoid the intermediate variable here.